Src/DependencyCollector/Shared/Implementation/DependencyTableStore.cs

```diff
@@ -29,8 +31,18 @@ internal static DependencyTableStore Instance
         {
            get 
            {
-              Interlocked.CompareExchange<DependencyTableStore>(ref instance, new DependencyTableStore(), null);
-              return instance;
+               if (instance == null)
+               {
+                   lock (SyncRoot)
+                   {
+                       if (instance == null)
```

Comment:
I don't recall, does C# suffer from problems with doubled checked locking the way that Java does?
Is there a reason not to just create the instance in the declaration and let the class loader handle it?